hazelcast/src/main/java/com/hazelcast/sql/impl/SqlServiceImpl.java

```diff
@@ -99,21 +104,31 @@ public SqlServiceImpl(NodeEngineImpl nodeEngine) {
         assert operationPoolSize > 0;
         assert queryTimeout >= 0L;
 
+        this.executorPoolSize = executorPoolSize;
+        this.operationPoolSize = operationPoolSize;
         this.queryTimeout = queryTimeout;
+    }
 
-        nodeServiceProvider = new NodeServiceProviderImpl(nodeEngine);
+    public void start() {
+        try {
+            jetSqlService = nodeEngine.getService(JetSqlService.SERVICE_NAME);
```

Comment:
Does Jet service have any state that needs to be cleaned or reinitialized on member stop or rejoin? If yes, we should consider adding callbacks to `reset` and `shutdown` methods.

Reply:
They are not needed at the moment but maybe it's safer to add them now

Reply:
I also think so. It is better to have no-op implementations now rather than waiting for the next IMDG release in case the risk becomes real.

Reply:
So it is done